parser/parser_go1.18.go

```diff
@@ -0,0 +1,61 @@
+//go:build go1.18
+// +build go1.18
+
+package parser
+
+import (
+	"go/ast"
+	"reflect"
+
+	"encr.dev/parser/est"
+	schema "encr.dev/proto/encore/parser/schema/v1"
+)
+
+func init() {
+	additionalTypeResolver = go118ResolveType
+}
+
+func go118ResolveType(p *parser, pkg *est.Package, file *est.File, expr ast.Expr) *schema.Type {
+	switch expr := expr.(type) {
+	// Needed for generic types with single generic parameters: `X[Index]` (i.e. `Vector[string]`)
+	case *ast.IndexExpr:
+		return resolveWithTypeArguments(p, pkg, file, expr.X, expr.Index)
+
+	// Needed for generic types with multiple generic parameters: `X[A, B]` (i.e. `Skiplist[string, string]`)
+	case *ast.IndexListExpr:
+		return resolveWithTypeArguments(p, pkg, file, expr.X, expr.Indices...)
+	}
+
+	return nil
+}
+
+func resolveWithTypeArguments(p *parser, pkg *est.Package, file *est.File, ident ast.Expr, typeArguments ...ast.Expr) *schema.Type {
+	// First we need to resolve the parameterized declaration
+	parameterizedType := p.resolveType(pkg, file, ident, nil)
+
+	named := parameterizedType.GetNamed()
+	if named == nil {
+		p.errf(ident.Pos(), "expected to get a named type, got %+v", reflect.TypeOf(parameterizedType.Typ))
+		panic(bailout{})
```

Comment:
We should only use `panic(bailout{})` if we need to abort the whole program parse; `p.errf` ensures that the parse will be marked as failed, but it allows for multiple errors to be returned which is helpful if there is more than one parsing error and you want to address them all at once.

In this case we could probably return a placeholder type instead of aborting the whole parse; possibly just `parameterizedType`/

Reply:
👍  I'll make the change, but I was deeming this as a "this shouldn't happen unless we've introduced a bug".